opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -868,9 +868,10 @@ namespace Opm
                 }
             }
             // injecting connections total volumerates at standard conditions
-            const Value factor = total_mob / volume_ratio;
             for (int componentIdx = 0; componentIdx < this->numComponents(); ++componentIdx) {
-                cq_s[componentIdx] = - Tw[componentIdx] * cmix_s[componentIdx] * factor * drawdown;
+                const Value cqt_i = - Tw[componentIdx] * (total_mob * drawdown);
+                Value cqt_is = cqt_i / volume_ratio;
+                cq_s[componentIdx] = cmix_s[componentIdx] * cqt_is;
             }
```

Comment:
For some reason the slight difference in the order of the operations to avoid computing the total_mob/volume_ratio factor for every component resulted in test failures in almost all the tests. I thus went back to the original order of operations.